Intermittent Blockstream Diff Testing for Previewnet Node failure

**Description:**
`sorted-export` command in `hedera-state-validator` can intermittently fail when running the current-version and next-version `sorted-export` JVMs in parallel.

Failure example [(link)](https://github.com/hashgraph/hedera-state-validator/actions/runs/28071371960/job/83106543796):

```text
java.lang.RuntimeException: java.lang.IllegalStateException: Initial capacity must be greater than 0, but was 0
	at com.hedera.statevalidation.util.StateUtils.initState(StateUtils.java:205)
	at com.hedera.statevalidation.util.StateUtils.getState(StateUtils.java:129)
	at com.hedera.statevalidation.util.StateUtils.getDefaultState(StateUtils.java:119)
	at com.hedera.statevalidation.SortedExportCommand.run(SortedExportCommand.java:71)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at com.hedera.statevalidation.StateOperatorCommand.main(StateOperatorCommand.java:48)
Caused by: java.lang.IllegalStateException: Initial capacity must be greater than 0, but was 0
	at com.swirlds.merkledb.MerkleDbDataSource.<init>(MerkleDbDataSource.java:339)
	at com.swirlds.merkledb.MerkleDbDataSourceBuilder.buildNewDataSource(MerkleDbDataSourceBuilder.java:106)
	at com.swirlds.merkledb.MerkleDbDataSourceBuilder.build(MerkleDbDataSourceBuilder.java:92)
	at com.swirlds.virtualmap.VirtualMap.<init>(VirtualMap.java:313)
	at com.swirlds.state.merkle.VirtualMapStateImpl.<init>(VirtualMapStateImpl.java:137)
	at com.swirlds.state.merkle.VirtualMapStateLifecycleManager.<init>(VirtualMapStateLifecycleManager.java:110)
	at com.hedera.statevalidation.util.StateUtils.initState(StateUtils.java:187)
	... 12 more
```

--- a/hedera-state-validator/src/main/java/com/hedera/statevalidation/StateOperatorCommand.java
+++ b/hedera-state-validator/src/main/java/com/hedera/statevalidation/StateOperatorCommand.java
@@ -37,6 +37,9 @@ public class StateOperatorCommand implements Runnable {
     /** Marker file written after a successful GCS download to distinguish complete from partial caches. */
     private static final String DOWNLOAD_COMPLETE_MARKER = ".download-complete";
 
+    private static final String STATE_DIR_PROPERTY = "state.dir";
+    private static final String TMP_DIR_PROPERTY = "tmp.dir";
+
     @Parameters(index = "0", description = "State directory. Accepts a local path or a GCS URI (gs://...).")
     private String stateDir;
 
@@ -71,7 +74,8 @@ public class StateOperatorCommand implements Runnable {
     void resolveAndGetStateDir() {
         if (resolvedStateDir != null) {
             // Already resolved (idempotent)
-            System.setProperty("state.dir", resolvedStateDir.getAbsolutePath());
+            System.setProperty(STATE_DIR_PROPERTY, resolvedStateDir.getAbsolutePath());
+            initializeTmpDirIfUnset();
             return;
         }
 
@@ -127,7 +131,17 @@ public class StateOperatorCommand implements Runnable {
             }
         }
 
-        System.setProperty("state.dir", resolvedStateDir.getAbsolutePath());
+        System.setProperty(STATE_DIR_PROPERTY, resolvedStateDir.getAbsolutePath());
+        initializeTmpDirIfUnset();
+    }
+
+    private void initializeTmpDirIfUnset() {
+        final String existingTmpDir = System.getProperty(TMP_DIR_PROPERTY, "");
+        if (!existingTmpDir.isBlank()) {
+            return;
+        }
+        System.setProperty(
+                TMP_DIR_PROPERTY, "state-validator-" + ProcessHandle.current().pid());
     }
 
     /**

--- a/hedera-state-validator/src/main/java/com/hedera/statevalidation/util/ConfigUtils.java
+++ b/hedera-state-validator/src/main/java/com/hedera/statevalidation/util/ConfigUtils.java
@@ -140,8 +140,9 @@ public final class ConfigUtils {
                 .withConverter(HederaFunctionalitySet.class, new FunctionalitySetConverter())
                 .withConverter(Bytes.class, new BytesConverter());
         if (!TMP_DIR.isEmpty()) {
-            configurationBuilder.withSource(
-                    new SimpleConfigSource().withValue("temporaryFiles.temporaryFilePath", TMP_DIR));
+            configurationBuilder.withSource(new SimpleConfigSource()
+                    .withValue("paths.tmpDir", TMP_DIR)
+                    .withValue("temporaryFiles.temporaryFilePath", TMP_DIR));
         }
         configuration = configurationBuilder.build();
     }

